Regressions in 4.14

## Issue Details

Some c++11/14 code has been integrated in 4.14 through bug fixes. Here is a list:

**Triangulation_examples**
 - [x] include/CGAL/NewKernel_d/Cartesian_filter_K.h:63: error: ‘constexpr’ does not name a type

 - [x] include/CGAL/typeset.h:73: error: no class template named ‘contains’ in ‘struct CGAL::Side_of_oriented_sphere_tag’

 - [x] include/CGAL/NewKernel_d/functor_tags.h:44: error: ‘value’ is not a member of ‘CGAL::typeset<CGAL::Orientation_of_points_tag, CGAL::Side_of_oriented_sphere_tag>::contains<CGAL::Compute_point_cartesian_coordinate_tag>’

 - [x] include/CGAL/NewKernel_d/Kernel_d_interface.h:59: error: no type named ‘type’ in ‘struct CGAL::Get_functor<CGAL::Cartesian_wrap<CGAL::Epick_d_help3<CGAL::Dimension_tag<4> >, CGAL::Epick_d<CGAL::Dimension_tag<4> > >, CGAL::Compute_point_cartesian_coordinate_tag, void, void>’

**libCGALCore_shared**
 - [x] /mnt/testsuite/include/CGAL/CORE/Filter.h:167:26: error: 'nan' is not a member of 'std'

**Scale_space_reconstruction_3_Examples**

 - [x] include/CGAL/Point_set_3.h:802:5: error: 'tie' is not a member of 'std'

 **Polygon_Demo**

 - [x] /usr/include/boost/utility/result_of.hpp:166:8: error: no class template named 'result' in 'const class CGAL::Construct_circulator_2<CGAL::Safe_circulator_from_iterator<__gnu_cxx::__normal_iterator<CGAL::Partition_vertex<CGAL::Partition_traits_2<CGAL::Epick> >*, std::vector<CGAL::Partition_vertex<CGAL::Partition_traits_2<CGAL::Epick> >, std::allocator<CGAL::Partition_vertex<CGAL::Partition_traits_2<CGAL::Epick> > > > >, int, int, int> >'
 struct result_of_nested_result : F::template result<FArgs>

 - [x] include/CGAL/Triangulation_2.h:283:3: error: no type named 'type' in 'struct boost::result_of<const CGAL::Construct_circulator_2<CGAL::Safe_circulator_from_iterator<__gnu_cxx::__normal_iterator<CGAL::Partition_vertex<CGAL::Partition_traits_2<CGAL::Epick> >*, std::vector<CGAL::Partition_vertex<CGAL::Partition_traits_2<CGAL::Epick> >, std::allocator<CGAL::Partition_vertex<CGAL::Partition_traits_2<CGAL::Epick> > > > >, int, int, int> >(const CGAL::Safe_circulator_from_iterator<__gnu_cxx::__normal_iterator<CGAL::Partition_vertex<CGAL::Partition_traits_2<CGAL::Epick> >*, std::vector<CGAL::Partition_vertex<CGAL::Partition_traits_2<CGAL::Epick> >, std::allocator<CGAL::Partition_vertex<CGAL::Partition_traits_2<CGAL::Epick> > > > >, int, int, int>&)>'

**Periodic_3_mesh_3**

 - [x] include/CGAL/Triangulation_3.h:561:54: error: could not convert '(& CGAL::Triangulation_3<GT, Tds, Lock_data_structure>::geom_traits<CGAL::Periodic_3_regular_triangulation_remove_traits_3<CGAL::Robust_periodic_weighted_circumcenter_traits_3<CGAL::Periodic_3_regular_triangulation_traits_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Periodic_3_offset_3, true> >, CGAL::Periodic_3_offset_3>, CGAL::Triangulation_data_structure_3<CGAL::Regular_triangulation_vertex_base_3<CGAL::Periodic_3_regular_triangulation_remove_traits_3<CGAL::Robust_periodic_weighted_circumcenter_traits_3<CGAL::Periodic_3_regular_triangulation_traits_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Periodic_3_offset_3, true> >, CGAL::Periodic_3_offset_3>, CGAL::Triangulation_ds_vertex_base_3<void> >, CGAL::Regular_triangulation_cell_base_3<CGAL::Periodic_3_regular_triangulation_remove_traits_3<CGAL::Robust_periodic_weighted_circumcenter_traits_3<CGAL::Periodic_3_regular_triangulation_traits_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Periodic_3_offset_3, true> >, CGAL::Periodic_3_offset_3>, CGAL::Triangulation_cell_base_3<CGAL::Periodic_3_regular_triangulation_remove_traits_3<CGAL::Robust_periodic_weighted_circumcenter_traits_3<CGAL::Periodic_3_regular_triangulation_traits_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Periodic_3_offset_3, true> >, CGAL::Periodic_3_offset_3>, CGAL::Triangulation_ds_cell_base_3<void> >, CGAL::Hidden_points_memory_policy<CGAL::Boolean_tag<true> >, std::list<std::pair<CGAL::Weighted_point_3<CGAL::Epick>, CGAL::Periodic_3_offset_3>, std::allocator<std::pair<CGAL::Weighted_point_3<CGAL::Epick>, CGAL::Periodic_3_offset_3> > > >, CGAL::Sequential_tag>, CGAL::Default>())->CGAL::Periodic_3_regular_triangulation_remove_traits_3<Gt_, Off_>::construct_point_3_object<CGAL::Robust_periodic_weighted_circumcenter_traits_3<CGAL::Periodic_3_regular_triangulation_traits_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Periodic_3_offset_3, true> >, CGAL::Periodic_3_offset_3>().CGAL::Construct_point_from_weighted_pair_3<Gt_, Construct_point_3_base_>::operator()<CGAL::Periodic_3_regular_triangulation_remove_traits_3<CGAL::Robust_periodic_weighted_circumcenter_traits_3<CGAL::Periodic_3_regular_triangulation_traits_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Periodic_3_offset_3, true> >, CGAL::Periodic_3_offset_3>, CGAL::Periodic_3_construct_point_3<CGAL::Periodic_3_triangulation_traits_base_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Periodic_3_offset_3>, CGAL::CartesianKernelFunctors::Construct_point_3<CGAL::Epick> > >((* & p))' from 'CGAL::Construct_point_from_weighted_pair_3<CGAL::Periodic_3_regular_triangulation_remove_traits_3<CGAL::Robust_periodic_weighted_circumcenter_traits_3<CGAL::Periodic_3_regular_triangulation_traits_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Periodic_3_offset_3, true> >, CGAL::Periodic_3_offset_3>, CGAL::Periodic_3_construct_point_3<CGAL::Periodic_3_triangulation_traits_base_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Periodic_3_offset_3>, CGAL::CartesianKernelFunctors::Construct_point_3<CGAL::Epick> > >::Point_3 {aka std::pair<CGAL::Point_3<CGAL::Epick>, CGAL::Periodic_3_offset_3>}' to 'CGAL::CartesianKernelFunctors::Construct_point_3<CGAL::Epick>::result<const CGAL::Construct_point_from_weighted_pair_3<CGAL::Periodic_3_regular_triangulation_remove_traits_3<CGAL::Robust_periodic_weighted_circumcenter_traits_3<CGAL::Periodic_3_regular_triangulation_traits_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Periodic_3_offset_3, true> >, CGAL::Periodic_3_offset_3>, CGAL::Periodic_3_construct_point_3<CGAL::Periodic_3_triangulation_traits_base_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Epick>, CGAL::Periodic_3_offset_3>, CGAL::CartesianKernelFunctors::Construct_point_3<CGAL::Epick> > >(const std::pair<CGAL::Weighted_point_3<CGAL::Epick>, CGAL::Periodic_3_offset_3>&)>::type {aka CGAL::Point_3<CGAL::Epick>}'

 **Nef_2**
 - [ ] include/CGAL/Nef_2/Polynomial.h:1226:16: internal compiler error: in generate_code_for_partition, at tree-loop-distribution.c:1323

**Kernel_23**
 - [x] /usr/include/boost/utility/result_of.hpp:166:8: error: no class template named 'result' in 'const class CGAL::Primitive_checker<CGAL::CartesianKernelFunctors::Construct_point_3<CGAL::Cartesian<CGAL::MP_Float> >, CGAL::CartesianKernelFunctors::Construct_point_3<CGAL::Cartesian<CGAL::MP_Float> >, CGAL::dont_check_equal>'
 struct result_of_nested_result : F::template result<FArgs>

**Number_types**
 - [x] /home/cgal_tester/build/src/cmake/platforms/CentOS7-Release/test/Number_types/CORE_Expr_ticket_4296.cpp:2:7: error: expected nested-name-specifier before 'NT'
 using NT = CORE::Expr;



## Environment

* CGAL version:4.14.4

--- a/CGAL_Core/include/CGAL/CORE/Filter.h
+++ b/CGAL_Core/include/CGAL/CORE/Filter.h
@@ -164,7 +164,6 @@ public:
       double maxVal = ( core_abs(val) + maxAbs / x.maxAbs) / xxx + DBL_MIN;
       return filteredFp(val, maxVal, 1 + core_max(ind, x.ind + 1));
     } else
-      //TODO: check that
       return filteredFp( std::strtod("NAN", (char**)NULL), getDoubleInfty(), 1);
   }
   /// square root

--- a/CGAL_Core/include/CGAL/CORE/Filter.h
+++ b/CGAL_Core/include/CGAL/CORE/Filter.h
@@ -164,7 +164,8 @@ public:
       double maxVal = ( core_abs(val) + maxAbs / x.maxAbs) / xxx + DBL_MIN;
       return filteredFp(val, maxVal, 1 + core_max(ind, x.ind + 1));
     } else
-      return filteredFp( std::nan(""), getDoubleInfty(), 1);
+      //TODO: check that
+      return filteredFp( std::strtod("NAN", (char**)NULL), getDoubleInfty(), 1);
   }
   /// square root
   filteredFp sqrt () const {

--- a/Installation/cmake/modules/CGAL_add_test.cmake
+++ b/Installation/cmake/modules/CGAL_add_test.cmake
@@ -16,6 +16,10 @@ if(NOT POLICY CMP0064)
   # Add a fake function to avoid CMake errors
   function(cgal_add_compilation_test)
   endfunction()
+  # Add a fake function to avoid CMake errors
+  function(cgal_setup_test_properties)
+  endfunction()
+  
 
   # Then return, to exit the file
   return()

--- a/Installation/include/CGAL/config.h
+++ b/Installation/include/CGAL/config.h
@@ -555,7 +555,7 @@ using std::max;
 #endif
 
 // Macro to specify [[no_unique_address]] if supported
-#if __has_cpp_attribute(no_unique_address)
+#if CGAL_CXX11 && __has_cpp_attribute(no_unique_address)
 #  define CGAL_NO_UNIQUE_ADDRESS [[no_unique_address]]
 #else
 #  define CGAL_NO_UNIQUE_ADDRESS

--- a/Kernel_23/include/CGAL/Kernel_checker.h
+++ b/Kernel_23/include/CGAL/Kernel_checker.h
@@ -88,6 +88,15 @@ public:
     : p1(pp1), p2(pp2), cmp(c)
   { }
 
+  template <typename F>
+  struct result;
+
+  template <typename F, typename A1>
+  struct result<F(A1)> {
+    typedef typename Pairify<typename CGAL::cpp11::result_of<P1(const A1&)>::type,
+                             typename CGAL::cpp11::result_of<P2(const A1&)>::type>::result_type type;
+  };
+
   template <class A1>
   typename Pairify<typename CGAL::cpp11::result_of<P1(const A1&)>::type,
                    typename CGAL::cpp11::result_of<P2(const A1&)>::type>::result_type
@@ -112,6 +121,12 @@ public:
     }
     return Pairify<result_type_1, result_type_2>()(res1, res2);
   }
+
+  template <typename F, typename A1, typename A2>
+  struct result<F(A1, A2)> {
+    typedef typename Pairify<typename CGAL::cpp11::result_of<P1(const A1&, const A2&)>::type,
+                             typename CGAL::cpp11::result_of<P2(const A1&, const A2&)>::type>::result_type type;
+  };
 
   template <class A1, class A2>
   typename Pairify<typename CGAL::cpp11::result_of<P1(const A1&, const A2&)>::type,
@@ -139,6 +154,12 @@ public:
     }
     return Pairify<result_type_1, result_type_2>()(res1, res2);
   }
+
+  template <typename F, typename A1, typename A2, typename A3>
+  struct result<F(A1, A2, A3)> {
+    typedef typename Pairify<typename CGAL::cpp11::result_of<P1(const A1&, const A2&, const A3&)>::type,
+                             typename CGAL::cpp11::result_of<P2(const A1&, const A2&, const A3&)>::type>::result_type type;
+  };
 
   template <class A1, class A2, class A3>
   typename Pairify<typename CGAL::cpp11::result_of<P1(
@@ -170,6 +191,16 @@ public:
     }
     return Pairify<result_type_1, result_type_2>()(res1, res2);
   }
+
+  template <typename F, typename A1, typename A2, typename A3, typename A4>
+  struct result<F(A1, A2, A3, A4)> {
+    typedef
+      typename Pairify<typename CGAL::cpp11::result_of<P1(
+                         const A1&, const A2&, const A3&, const A4&)>::type,
+                       typename CGAL::cpp11::result_of<P2(
+                         const A1&, const A2&, const A3&, const A4&)>::type>::result_type
+    type;
+  };
 
   template <class A1, class A2, class A3, class A4>
   typename Pairify<typename CGAL::cpp11::result_of<P1(
@@ -206,6 +237,16 @@ public:
     }
     return Pairify<result_type_1, result_type_2>()(res1, res2);
   }
+
+  template <typename F, typename A1, typename A2, typename A3, typename A4, typename A5>
+  struct result<F(A1, A2, A3, A4, A5)> {
+    typedef
+      typename Pairify<typename CGAL::cpp11::result_of<P1(
+                         const A1&, const A2&, const A3&, const A4&, const A5&)>::type,
+                       typename CGAL::cpp11::result_of<P2(
+                         const A1&, const A2&, const A3&, const A4&, const A5&)>::type>::result_type
+    type;
+  };
 
   template <class A1, class A2, class A3, class A4, class A5>
   typename Pairify<typename CGAL::cpp11::result_of<P1(

--- a/NewKernel_d/include/CGAL/NewKernel_d/Cartesian_filter_K.h
+++ b/NewKernel_d/include/CGAL/NewKernel_d/Cartesian_filter_K.h
@@ -60,8 +60,8 @@ struct Cartesian_filter_K : public Base_
     CGAL_NO_UNIQUE_ADDRESS Store_kernel<AK_> sak;
     CGAL_NO_UNIQUE_ADDRESS Store_kernel<EK_> sek;
 
-    constexpr Cartesian_filter_K(){}
-    constexpr Cartesian_filter_K(int d):Base_(d){}
+    CGAL_CONSTEXPR Cartesian_filter_K(){}
+    CGAL_CONSTEXPR Cartesian_filter_K(int d):Base_(d){}
     //FIXME: or do we want an instance of AK and EK belonging to this kernel,
     //instead of a reference to external ones?
     CGAL_CONSTEXPR Cartesian_filter_K(AK_ const&a,EK_ const&b):Base_(),sak(a),sek(b){}

--- a/NewKernel_d/include/CGAL/NewKernel_d/Cartesian_filter_K.h
+++ b/NewKernel_d/include/CGAL/NewKernel_d/Cartesian_filter_K.h
@@ -32,22 +32,28 @@ namespace CGAL {
   // It would be nicer to write the table in the other direction: Orientation_of_points_tag is good up to 6, Side_of_oriented_sphere_tag up to 5, etc.
 template<class> struct Functors_without_division { typedef typeset<> type; };
 template<> struct Functors_without_division<Dimension_tag<1> > {
-  typedef typeset<Orientation_of_points_tag, Side_of_oriented_sphere_tag> type;
+  typedef typeset<Orientation_of_points_tag>
+  ::add<Side_of_oriented_sphere_tag>::type type;
 };
 template<> struct Functors_without_division<Dimension_tag<2> > {
-  typedef typeset<Orientation_of_points_tag, Side_of_oriented_sphere_tag> type;
+  typedef typeset<Orientation_of_points_tag>
+  ::add<Side_of_oriented_sphere_tag>::type type;
 };
 template<> struct Functors_without_division<Dimension_tag<3> > {
-  typedef typeset<Orientation_of_points_tag, Side_of_oriented_sphere_tag> type;
+  typedef typeset<Orientation_of_points_tag>
+  ::add<Side_of_oriented_sphere_tag >::type  type;
 };
 template<> struct Functors_without_division<Dimension_tag<4> > {
-  typedef typeset<Orientation_of_points_tag, Side_of_oriented_sphere_tag> type;
+  typedef typeset<Orientation_of_points_tag>
+  ::add<Side_of_oriented_sphere_tag>::type type;
 };
 template<> struct Functors_without_division<Dimension_tag<5> > {
-  typedef typeset<Orientation_of_points_tag, Side_of_oriented_sphere_tag> type;
+  typedef typeset<Orientation_of_points_tag>
+  ::add<Side_of_oriented_sphere_tag>::type type;
 };
 template<> struct Functors_without_division<Dimension_tag<6> > {
-  typedef typeset<Orientation_of_points_tag, Side_of_oriented_sphere_tag> type;
+  typedef typeset<Orientation_of_points_tag>
+  ::add<Side_of_oriented_sphere_tag>::type type;
 };
 
 // FIXME:

--- a/Number_types/test/Number_types/CORE_Expr_ticket_4296.cpp
+++ b/Number_types/test/Number_types/CORE_Expr_ticket_4296.cpp
@@ -1,5 +1,5 @@
 #include <CGAL/CORE/Expr.h>
-using NT = CORE::Expr;
+typedef CORE::Expr NT;
 
 int main() {
 

--- a/Partition_2/include/CGAL/Partition_2/Triangulation_indirect_traits_2.h
+++ b/Partition_2/include/CGAL/Partition_2/Triangulation_indirect_traits_2.h
@@ -122,6 +122,7 @@ template <class Circulator>
 class Construct_circulator_2
 {
 public:
+   typedef Circulator result_type;
    Circulator operator()(Circulator p1) const { return p1; }
 };
 

--- a/Periodic_3_triangulation_3/include/CGAL/internal/Periodic_3_Delaunay_triangulation_remove_traits_3.h
+++ b/Periodic_3_triangulation_3/include/CGAL/internal/Periodic_3_Delaunay_triangulation_remove_traits_3.h
@@ -49,6 +49,14 @@ public:
 
   using Base::operator();
 
+  template<typename F>
+  struct result : Base::template result<F> {};
+
+  template<typename F>
+  struct result<F(Point_3)> {
+    typedef const Point_3& type;
+  };
+
   const Point_3& operator()(const Point_3& p) const { return p; }
 };
 

--- a/Periodic_3_triangulation_3/include/CGAL/internal/Periodic_3_regular_triangulation_remove_traits_3.h
+++ b/Periodic_3_triangulation_3/include/CGAL/internal/Periodic_3_regular_triangulation_remove_traits_3.h
@@ -58,6 +58,14 @@ public:
 
   using Base::operator(); // for K::Weighted_point_3 to Point_3
 
+  template<typename F>
+  struct result : Base::template result<F> {};
+
+  template<typename F>
+  struct result<F(Weighted_point_3)> {
+    typedef Point_3 type;
+  };
+
   Point_3 operator()(const Weighted_point_3& wp) const {
     return std::make_pair(operator()(wp.first), wp.second /* offset */);
   }

--- a/Point_set_3/include/CGAL/Point_set_3.h
+++ b/Point_set_3/include/CGAL/Point_set_3.h
@@ -799,7 +799,7 @@ public:
   {
     Property_map<T> pm;
     bool added = false;
-    std::tie (pm, added) = m_base.template add<T> (name, t);
+    boost::tie (pm, added) = m_base.template add<T> (name, t);
     return std::make_pair (pm, added);
   }
 
@@ -820,7 +820,7 @@ public:
   {
     Property_map<T> pm;
     bool okay = false;
-    std::tie (pm, okay) = m_base.template get<T>(name);
+    boost::tie (pm, okay) = m_base.template get<T>(name);
     return std::make_pair (pm, okay);
   }
 
@@ -864,7 +864,7 @@ public:
   std::pair<Vector_map, bool> add_normal_map (const Vector& default_value = CGAL::NULL_VECTOR)
   {
     bool out = false;
-    std::tie (m_normals, out) = this->add_property_map<Vector> ("normal", default_value);
+    boost::tie (m_normals, out) = this->add_property_map<Vector> ("normal", default_value);
     return std::make_pair (m_normals, out);
   }
   /*!

--- a/STL_Extension/include/CGAL/Concurrent_compact_container.h
+++ b/STL_Extension/include/CGAL/Concurrent_compact_container.h
@@ -47,11 +47,6 @@
 
 #include <boost/mpl/if.hpp>
 
-#ifdef CGAL_CXX11
-#define CGAL_CONSTEXPR constexpr
-#else
-#define CGAL_CONSTEXPR
-#endif
 namespace CGAL {
 
 #define CGAL_GENERATE_MEMBER_DETECTOR(X)                                           \

--- a/STL_Extension/include/CGAL/Concurrent_compact_container.h
+++ b/STL_Extension/include/CGAL/Concurrent_compact_container.h
@@ -47,6 +47,11 @@
 
 #include <boost/mpl/if.hpp>
 
+#ifdef CGAL_CXX11
+#define CGAL_CONSTEXPR constexpr
+#else
+#define CGAL_CONSTEXPR
+#endif
 namespace CGAL {
 
 #define CGAL_GENERATE_MEMBER_DETECTOR(X)                                           \
@@ -192,7 +197,7 @@ protected:
 
 #if CGAL_CONCURRENT_COMPACT_CONTAINER_APPROXIMATE_SIZE
   // `m_size` plus or minus `precision_of_approximate_size - 1`
-  static constexpr double precision_of_approximate_size_plus_1 = 1.10;
+  static CGAL_CONSTEXPR double precision_of_approximate_size_plus_1 = 1.10;
   size_type m_approximate_size;
   std::atomic<size_type> m_atomic_approximate_size;
   void refresh_approximate_size() {

--- a/STL_Extension/include/CGAL/Concurrent_compact_container.h
+++ b/STL_Extension/include/CGAL/Concurrent_compact_container.h
@@ -149,14 +149,14 @@ public:
   void inc_size() {
     ++m_size;
 #if CGAL_CONCURRENT_COMPACT_CONTAINER_APPROXIMATE_SIZE
-    if(m_size > (m_approximate_size * precision_of_approximate_size_plus_1))
+    if(m_size > (m_approximate_size * 1.10))
       refresh_approximate_size();
 #endif // CGAL_CONCURRENT_COMPACT_CONTAINER_APPROXIMATE_SIZE
   }
   void dec_size() {
     --m_size;
 #if CGAL_CONCURRENT_COMPACT_CONTAINER_APPROXIMATE_SIZE
-    if((m_size * precision_of_approximate_size_plus_1) < m_approximate_size)
+    if((m_size * 1.10) < m_approximate_size)
       refresh_approximate_size();
 #endif // CGAL_CONCURRENT_COMPACT_CONTAINER_APPROXIMATE_SIZE
   }
@@ -197,7 +197,6 @@ protected:
 
 #if CGAL_CONCURRENT_COMPACT_CONTAINER_APPROXIMATE_SIZE
   // `m_size` plus or minus `precision_of_approximate_size - 1`
-  static CGAL_CONSTEXPR double precision_of_approximate_size_plus_1 = 1.10;
   size_type m_approximate_size;
   std::atomic<size_type> m_atomic_approximate_size;
   void refresh_approximate_size() {

--- a/Triangulation_3/include/CGAL/Triangulation_3.h
+++ b/Triangulation_3/include/CGAL/Triangulation_3.h
@@ -555,7 +555,7 @@ protected:
 
 public:
   template<typename P> // Point or Point_3
-  Point_3
+  typename boost::result_of<Construct_point_3(P)>::type
   construct_point(const P& p) const
   {
     return geom_traits().construct_point_3_object()(p);

--- a/Triangulation_3/include/CGAL/Triangulation_3.h
+++ b/Triangulation_3/include/CGAL/Triangulation_3.h
@@ -555,7 +555,7 @@ protected:
 
 public:
   template<typename P> // Point or Point_3
-  typename boost::result_of<const Construct_point_3(const P&)>::type
+  Point_3
   construct_point(const P& p) const
   {
     return geom_traits().construct_point_3_object()(p);

